marlowe-dashboard-client/src/InputField/View.purs

```diff
@@ -7,16 +7,27 @@ import Data.Foldable (foldMap)
 import Data.Lens (view)
 import Data.Maybe (Maybe(..), isJust)
 import Data.String (Pattern(..), contains, toLower)
+import Halogen (RefLabel(..))
 import Halogen.Css (classNames)
 import Halogen.HTML (HTML, a, div, div_, input, span_, text)
 import Halogen.HTML.Events (onBlur, onMouseEnter, onMouseLeave)
 import Halogen.HTML.Events.Extra (onBlur_, onClick_, onFocus_, onValueInput_)
-import Halogen.HTML.Properties (InputType(..), autocomplete, id_, placeholder, readOnly, type_, value)
+import Halogen.HTML.Properties (InputType(..), autocomplete, id_, placeholder, readOnly, ref, tabIndex, type_, value)
 import InputField.Lenses (_additionalCss, _dropdownLocked, _dropdownOpen, _id_, _placeholder, _pristine, _readOnly, _value)
 import InputField.State (validate)
 import InputField.Types (class InputFieldError, Action(..), InputDisplayOptions, State, inputErrorToString)
 import Marlowe.Extended.Metadata (NumberFormat(..))
 
+inputCss :: InputDisplayOptions -> Boolean -> Array String
+inputCss { readOnly: true } = Css.inputNoFocus
+
+inputCss _ = Css.input
+
+getTabIndex :: InputDisplayOptions -> Int
+getTabIndex { readOnly: true } = -1
+
+getTabIndex _ = 0
```

Comment:
not too familiar with tab index, but shouldn't we specify with incremental order? 0, 1, 2, etc?

Reply:
tab index `0` means the element will use the default tab order assigned by the browser. `-1` means that it is not focusable by tabbing.

Reply:
We could make this an ADT to make it more explicit...

Reply:
But that might make the semantics difficult to line up with... I think leaving as `Int` is the best, the semantics are web standard and generally well undeerstood https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/tabindex